### PR TITLE
Add additional error classes and error codes

### PR DESCRIPTION
**Pull request by gaojude**
Adds support for additional error classes in the error code SWC plugin.

Follow up for https://github.com/vercel/next.js/pull/73332#discussion_r1873491364

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| buildDuration | 21.9s | 19.2s | N/A |
| buildDurationCached | 18.4s | 15.4s | N/A |
| nodeModulesSize | 410 MB | 410 MB | ⚠️ +163 kB |
| nextStartRea..uration (ms) | 547ms | 521ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| 1187-HASH.js gzip | 50.8 kB | 50.9 kB | N/A |
| 8276.HASH.js gzip | 169 B | 168 B | N/A |
| 8377-HASH.js gzip | 5.36 kB | 5.36 kB | N/A |
| bccd1874-HASH.js gzip | 53 kB | 53 kB | N/A |
| framework-HASH.js gzip | 57.5 kB | 57.5 kB | N/A |
| main-app-HASH.js gzip | 232 B | 235 B | N/A |
| main-HASH.js gzip | 34 kB | 34 kB | N/A |
| webpack-HASH.js gzip | 1.71 kB | 1.71 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 39.4 kB | 39.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| _app-HASH.js gzip | 193 B | 193 B | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 512 B | 510 B | N/A |
| css-HASH.js gzip | 343 B | 342 B | N/A |
| dynamic-HASH.js gzip | 1.84 kB | 1.84 kB | ✓ |
| edge-ssr-HASH.js gzip | 265 B | 265 B | ✓ |
| head-HASH.js gzip | 363 B | 362 B | N/A |
| hooks-HASH.js gzip | 393 B | 392 B | N/A |
| image-HASH.js gzip | 4.49 kB | 4.49 kB | N/A |
| index-HASH.js gzip | 268 B | 268 B | ✓ |
| link-HASH.js gzip | 2.35 kB | 2.34 kB | N/A |
| routerDirect..HASH.js gzip | 328 B | 328 B | ✓ |
| script-HASH.js gzip | 397 B | 397 B | ✓ |
| withRouter-HASH.js gzip | 323 B | 326 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.59 kB | 3.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| _buildManifest.js gzip | 747 B | 746 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| index.html gzip | 524 B | 522 B | N/A |
| link.html gzip | 539 B | 537 B | N/A |
| withRouter.html gzip | 520 B | 519 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| edge-ssr.js gzip | 128 kB | 128 kB | ⚠️ +114 B |
| page.js gzip | 203 kB | 204 kB | ⚠️ +314 B |
| Overall change | 332 kB | 332 kB | ⚠️ +428 B |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 670 B | 667 B | N/A |
| middleware-r..fest.js gzip | 155 B | 156 B | N/A |
| middleware.js gzip | 31.2 kB | 31.2 kB | N/A |
| edge-runtime..pack.js gzip | 844 B | 844 B | ✓ |
| Overall change | 844 B | 844 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| 523-experime...dev.js gzip | 322 B | 322 B | ✓ |
| 523.runtime.dev.js gzip | 314 B | 314 B | ✓ |
| app-page-exp...dev.js gzip | 323 kB | 323 kB | ⚠️ +275 B |
| app-page-exp..prod.js gzip | 127 kB | 127 kB | ⚠️ +168 B |
| app-page-tur..prod.js gzip | 140 kB | 140 kB | ⚠️ +160 B |
| app-page-tur..prod.js gzip | 135 kB | 135 kB | ⚠️ +158 B |
| app-page.run...dev.js gzip | 313 kB | 313 kB | ⚠️ +269 B |
| app-page.run..prod.js gzip | 123 kB | 123 kB | ⚠️ +167 B |
| app-route-ex...dev.js gzip | 37.3 kB | 37.4 kB | ⚠️ +106 B |
| app-route-ex..prod.js gzip | 25.4 kB | 25.5 kB | ⚠️ +101 B |
| app-route-tu..prod.js gzip | 25.4 kB | 25.5 kB | ⚠️ +101 B |
| app-route-tu..prod.js gzip | 25.2 kB | 25.3 kB | ⚠️ +104 B |
| app-route.ru...dev.js gzip | 38.9 kB | 39 kB | N/A |
| app-route.ru..prod.js gzip | 25.2 kB | 25.3 kB | ⚠️ +104 B |
| pages-api-tu..prod.js gzip | 9.67 kB | 9.68 kB | N/A |
| pages-api.ru...dev.js gzip | 11.6 kB | 11.6 kB | N/A |
| pages-api.ru..prod.js gzip | 9.66 kB | 9.68 kB | N/A |
| pages-turbo...prod.js gzip | 21.7 kB | 21.7 kB | N/A |
| pages.runtim...dev.js gzip | 27.4 kB | 27.4 kB | N/A |
| pages.runtim..prod.js gzip | 21.7 kB | 21.7 kB | N/A |
| server.runti..prod.js gzip | 916 kB | 916 kB | N/A |
| Overall change | 1.3 MB | 1.3 MB | ⚠️ +1.71 kB |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js jude/more-error-classes | Change |
| - | - | - | - |
| 0.pack gzip | 2.05 MB | 2.05 MB | ⚠️ +1.73 kB |
| index.pack gzip | 72 kB | 72.5 kB | ⚠️ +473 B |
| Overall change | 2.12 MB | 2.12 MB | ⚠️ +2.2 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>1187-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-ex..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-route.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api-tu..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-api.ru..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages-turbo...time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>pages.runtime.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: a8a668f4bd1ab42e394ff30f6bfbf96f2c3e04f1

**Comment by gaojude** [comment]
* **#73865** <a href="https://app.graphite.dev/github/pr/vercel/next.js/73865?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#73862** <a href="https://app.graphite.dev/github/pr/vercel/next.js/73862?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/next.js/73862?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by ijjk** [comment]
## Failing test suites 
Commit: fa50fbd44209ca665ea3db765c4dae35a9d312bb

`__NEXT_EXPERIMENTAL_PPR=true pnpm test-start test/e2e/app-dir/missing-suspense-with-csr-bailout/missing-suspense-with-csr-bailout.test.ts` (PPR)
- missing-suspense-with-csr-bailout > useSearchParams > should fail build if useSearchParams is not wrapped in a suspense boundary
- missing-suspense-with-csr-bailout > useSearchParams > should pass build if useSearchParams is wrapped in a suspense boundary

<details>
<summary>Expand output</summary>

  ● missing-suspense-with-csr-bailout › useSearchParams › should fail build if useSearchParams is not wrapped in a suspense boundary

    expect(received).toContain(expected) // indexOf

    Expected substring: "useSearchParams() should be wrapped in a suspense boundary at page \"/\"."
    Received string:    "   Loading config from /tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/next.config.js
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
       Loading config from /tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/next.config.js
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
       ▲ Next.js 15.1.1-canary.1·
       Creating an optimized production build ...
       Loading config from /tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/next.config.js
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
       Loading config from /tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/next.config.js
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
       Loading config from /tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/next.config.js
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
     ✓ Compiled successfully
       Linting and checking validity of types ...
       Collecting page data ...
       Generating static pages (0/5) ...
    Error occurred prerendering page \"/\". Read more: https://nextjs.org/docs/messages/prerender-error
    Error: Could not find the module \"/tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/app/page.js#default\" in the React Client Manifest. This is probably a bug in the React Server Components bundler.

      28 |       const { exitCode } = await next.build()
      29 |       expect(exitCode).toBe(1)
    > 30 |       expect(next.cliOutput).toContain(message)
         |                              ^
      31 |       // Can show the trace where the searchParams hook is used
      32 |       expect(next.cliOutput).toMatch(/at.*server[\\/]app[\\/]page.js/)
      33 |     })

      at eD (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:16632)
      at eH (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:24311)
      at ../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:15445
      at Object.toJSON (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:16132)
          at stringify (<anonymous>)
      at eY (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:27927)
      at eQ (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:28157)
      at eZ (../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:28849)
      at ../../../../../../tmp/next-install-39d357757a2ac7a7e4500d72efb69fbb59aee8dba8c41fd21d069175d0a432e7/node_modules/.pnpm/next@file+..+next-repo-244fb7ffb66aff0077b70ea58ed2edd9a019f93bb565729f0486c977226a7a5c+packa_7icb2vtbgkl4l725xlepr3t4ay/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:116:29746
      Export encountered an error on /page: /, exiting the build.
       ⨯ Static worker exited with code: 1 and signal: null
      "
      at Object.toContain (e2e/app-dir/missing-suspense-with-csr-bailout/missing-suspense-with-csr-bailout.test.ts:30:30)

  ● missing-suspense-with-csr-bailout › useSearchParams › should pass build if useSearchParams is wrapped in a suspense boundary

    expect(received).resolves.toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

      Object {
        "cliOutput": StringNotContaining "useSearchParams() should be wrapped in a suspense boundary at page \"/\".",
    -   "exitCode": 0,
    +   "exitCode": 1,
      }

      37 |       await next.renameFile('app/layout-suspense.js', 'app/layout.js')
      38 |
    > 39 |       await expect(next.build()).resolves.toEqual({
         |                                           ^
      40 |         exitCode: 0,
      41 |         cliOutput: expect.not.stringContaining(message),
      42 |       })

      at Object.toEqual (../node_modules/.pnpm/expect@29.7.0/node_modules/expect/build/index.js:174:22)
      at Object.toEqual (e2e/app-dir/missing-suspense-with-csr-bailout/missing-suspense-with-csr-bailout.test.ts:39:43)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by devjiwonchoi — APPROVED** [review]

File: crates/next-error-code-swc-plugin/src/lib.rs [hunk 1]
```diff
@@ -16,8 +16,43 @@ struct NewError {
 }
 
 fn is_error_class_name(name: &str) -> bool {
-    // TODO: handle other error classes
-    name == "Error"
+    name == "AggregateError"
```
Comment (eps1lon) [inline crates/next-error-code-swc-plugin/src/lib.rs:19]: We should document how you gathered these.

**Pull request merged**